pkg/machinery/config/types/v1alpha1/v1alpha1_config_schema_test.go

```diff
@@ -0,0 +1,128 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package v1alpha1_test
+
+import (
+	"bytes"
+	_ "embed"
+	"testing"
+	"text/template"
+
+	validatejsonschema "github.com/santhosh-tekuri/jsonschema/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+type templateOptions struct {
+	Version                     string
+	Persist                     string
+	MachineType                 string
+	MachineTimeBootTimeout      string
+	ClusterControlPlaneEndpoint string
+}
+
+var (
+	//go:embed v1alpha1_config.schema.json
+	schemaData string
+
+	//go:embed testdata/schema/v1alpha1_config.tmpl.yaml
+	configTemplateData []byte
+)
+
+func TestSchemaValidation(t *testing.T) {
+	schema, err := validatejsonschema.CompileString("https://www.talos.dev/schemas/v1alpha1/config.json", schemaData)
```

Comment:
my only concern with this test is how we can keep it up to date, as it is based on static machine config. 

can we probably try to use `generate` package here and generate the machine config on the fly with the test, patching it as necessary for each test case?